add support for licenses

dirk-thomas opened the pull request:
Combined copyright and license checking.

Currently implemented entry points:
* copyright holders:
  * OSRF 
* licenses
  * Apache2

Connects to ament/ament_lint#10

dirk-thomas commented:
I will keep iterating on this while processing the current repos:

* https://github.com/ament/ament_package/commit/889a39037870b6f8543beea810fbf2a5ea738798

dirk-thomas commented:
The current state already worked for eight repository. So I think it is good enough.

@esteve @tfoote @wjwwood Please review.

dirk-thomas commented:
@esteve @tfoote @wjwwood Please review.

esteve commented on ament_copyright/ament_copyright/licenses.py:
```diff
@@ -0,0 +1,260 @@
+See the License for the specific language governing permissions and
+limitations under the License.""",
+
+    """\
```
Wouldn't it be better to move the text of the license to a separate file and read it?

esteve commented on ament_copyright/ament_copyright/licenses.py:
```diff
@@ -0,0 +1,260 @@
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   {copyright}
```
The text of the Apache license is strongly copyrighted, so I guess any modifications would make it invalid. This is part of the appendix, though, so we're probably safe, but I wouldn't replace the original sentence (i.e. "Copyright [yyyy] [name of copyright owner]").

dirk-thomas commented:
@esteve I moved the apache2 snippets to separate files and modified the behavior to *not* replace the copyright template in the LICENSE files. All repos have been updated accordingly.

esteve commented on ament_copyright/doc/index.rst:
```diff
@@ -2,11 +2,18 @@ ament_copyright
 
 Checks C / C++ / CMake / Python source files for the existance of a copyright
-notice.
+and licence notice.
```
s/licence/license/

esteve commented:
+1

wjwwood commented:
The changes to the other repositories look good to me, still reviewing the code in this pr.

wjwwood commented:
Looks good to me. +1

tfoote commented:
lgtm
